src/whenever/__init__.py

```diff
@@ -2499,18 +2526,24 @@ def from_rfc2822(cls, s: str, /) -> UTCDateTime:
           Use :meth:`OffsetDateTime.from_rfc2822` if you'd like to
           parse an RFC 2822 string with a nonzero offset.
         """
-        parsed = _parse_rfc2822(s)
-        # Nested ifs to keep happy path fast
-        if parsed.tzinfo is not _UTC:
-            if parsed.tzinfo is None:
-                if "-0000" not in s:
-                    raise ValueError("RFC 2822 string must have a UTC offset")
-                parsed = parsed.replace(tzinfo=_UTC)
-            else:
-                raise ValueError(
-                    "RFC 2822 string can't have nonzero offset to be parsed as UTC"
-                )
-        return cls._from_py_unchecked(parsed)
+        try:
+            parsed = _parse_rfc2822(s)
+            # Nested ifs to keep happy path fast
+            if parsed.tzinfo is not _UTC:
+                if parsed.tzinfo is None:
+                    if "-0000" not in s:
+                        raise ValueError(
+                            "RFC 2822 string must have a UTC offset"
+                        )
+                    parsed = parsed.replace(tzinfo=_UTC)
+                else:
+                    raise ValueError(
+                        "RFC 2822 string can't have nonzero offset "
+                        "to be parsed as UTC"
+                    )
+            return cls._from_py_unchecked(parsed)
+        except ValueError as e:
+            raise ValueError(f"Cannot parse as RFC 2822 string: {s!r}") from e
```

Comment:
chaining value errors like this—interesting approach 👍  (no change needed)

Reply:
Thanks! This is the only instance of exception chaining I left. I removed the others because they either chained on more specific errors I had added to this PR originally but have now removed, or chained on errors from functions or methods on which they delegated, but didn't add more information.

The generation of more specific error messages in this function might be inconsistent with the intention to stick to one generic error (insofar the chained errors are considered part of the error, or at least the error context), but leaving it here as it predates this PR.